985 Template sub params

JChen2005 opened the pull request:
Now allows user to retrieve child param even when parent param has no values.

JChen2005 committed:
Added an extra condition in intializeParamWithValue to allow template params

abrown-RV reviewed (changes requested):
This solution gives the struct a value. I believe this is breaking the intention of the device. If a param does not have a value we should not give it one. This is probably not a codegen problem, but a problem with the getParam function.

Potentially a discussion for tomorrow's checkin.

JChen2005 committed:
Added statements to check for template params

JChen2005 committed:
Reverted changes in codegen

NR-RV commented on sdks/cpp/common/include/ParamWithValue.h:
update the Copyright year and file meta data
line 4 and 33-40

NR-RV replied:
and add your name to the authors and update the date

NR-RV commented on sdks/cpp/common/src/Device.cpp:
```diff
@@ -558,5 +588,4 @@ bool Device::shouldSendParam(const IParam& param, bool is_subscribed, const IAut
     return should_send;
-}
-
+}
```
watch what your modifing
dont let the AI write to the full file

NR-RV reviewed (changes requested):
check things befor asking for review :)

JChen2005 committed:
Fixed a ctest

JChen2005 committed:
Reverted changes made in unit tests, Added back missing comments

JChen2005 committed:
Added tge getParam functuon, updated copyright

nelgd committed:
Asset req test #820 (#1045)

Closes issue #820

- Created and Modified files for new ExternalObjectRequest unit tests
- Updated launch.json to include ExternalObjectRequest test
- Updated UT fixture and added some tests.
- Added some more unit tests. Works, but needs some tweaking.
- Added more unit tests
- Compacted unit tests + Rebase
- Revised Unit Tests based on feedback.
- Simplified cleanup function.
- Made default case ignored by coverage.

abrown-RV committed:
938 disconnect issues (#1012)

Closes #938

Adds heartbeat to keep the connection alive

JChen2005 committed:
Added a test to check if empty child param is empty

JChen2005 committed:
Removed test 13

JChen2005 committed:
Added two test to check for nesting

NR-RV committed:
first attempt at redoing the cmake (#1050)

* first attempt at redoing the cmake

* got the unit tests working agian

* remove thedocker stuff and add thecorrect depencey for the openaip buld thing

* dont need to always build the examples

* add the bare  to common

* add the building of examples back to CodeQL

* remove vdk form coverage cuz its not ours

* remove vdk from the workflow gcover

JChen2005 committed:
Fixed indenting

JChen2005 committed:
Fixed indenting

JChen2005 committed:
removed a header file being included

JChen2005 committed:
Reverted some changes

JChen2005 committed:
Added 2nd unit test

NR-RV commented on sdks/cpp/common/src/Device.cpp:
dont change files you didnt touch

NR-RV commented on tools/codegen/cpp/cppgen.js:
dont touch files you didnt edit

NR-RV commented on unittests/cpp/common/tests/ParamWithValue_test/ParamWithValue_test.cpp:
```diff
@@ -270,3 +260,17 @@ TEST_F(ParamWithValueTest, ParamInfoToProto_Error) {
         << "toProto should return PERMISSION_DENIED if Authorizer does not have readAuthz.";
 }
+
+TEST_F(ParamWithValueTest, EmptyChild) {
```
bad test name and needs a comment to explain what it dose

NR-RV commented on unittests/cpp/common/tests/ParamWithValue_test/ParamWithValue_test.cpp:
```diff
@@ -270,3 +260,17 @@ TEST_F(ParamWithValueTest, ParamInfoToProto_Error) {
+    // Verify the child field has the expected empty/default value
+    EXPECT_EQ(getParamValue<int32_t>(foundParam.get()), 0) 
+        << "Empty child field should have value 0";
+}
```
what happend to the other test you had for testing a param that has a child with a child?

NR-RV commented on unittests/cpp/common/CommonTestHelpers.h:
update the meta data on line 31-36

NR-RV commented on unittests/cpp/common/tests/ParamWithValue_test/ParamWithValue_test.cpp:
update meta data on line 31-37

JChen2005 committed:
Finished All three unit tests, added updated comments and names
